esmvaltool/recipes/recipe_modes_of_variability_wp4.yml

```diff
@@ -0,0 +1,79 @@
+###############################################################################
+# recipe_modes_of_variability.yml
+#
+# Description
+# Tool to compute the RMSE between the observed and modelled patterns of
+# variability obtained through classification and their relative relative bias
+#(percentage) in the frequency of occurrence and the persistence of each mode.
+#
+#
+# Authors
+# Verónica Torralba
+# Neven Fuckar
+# Nicola Cortesi
+# Virginie Guemas
+# Alasdair Hunter
+# Nicolau Manubens
+# Núria Pérez-Zanón
+#
+# Project
+# KNMI Magic
+#
+# Reference
+# Fučkar, Neven S., Virginie Guemas, Nathaniel C. Johnson, François Massonnet,
+# and Francisco J. Doblas-Reyes. "Clusters of interannual sea ice variability in
+# the northern hemisphere." Climate Dynamics 47, no. 5-6 (2016): 1527-1543.
+#
+#
+# This namelist is part of the ESMValTool
+###############################################################################
+---
+
+datasets:
+  - {dataset: MIROC5, type: exp,  project: CMIP5,  exp: historical,  ensemble: i0p1,  start_year: 1981,  end_year: 1990}
+  - {dataset: MIROC5, type: exp,  project: CMIP5,  exp: rcp85,  ensemble: r1i1p1,  start_year: 2006,  end_year: 2026}
+  #- {dataset: bcc-csm1-1, type: exp,  project: CMIP5,  mip: day,  exp: historical,  ensemble: r1i1p1,  start_year: 1961,  end_year: 1990}
+  #- {dataset: bcc-csm1-1, type: exp,  project: CMIP5,  mip: day,  exp: rcp26,  ensemble: r1i1p1,  start_year: 2020,  end_year: 2040}
+  #- {dataset: IPSL-CM5A-LR,  type: exp, project: CMIP5,  mip: day,  exp: historical,  ensemble: r1i1p1,  start_year: 1961,  end_year: 1990}
+  #- {dataset: MPI-ESM-LR, type: exp, project: CMIP5,  exp: historical,  ensemble: r1i1p1,  start_year: 1961,  end_year: 1990}
+  #- {dataset: MPI-ESM-LR, type: exp, project: CMIP5,  exp: rcp85,  ensemble: r1i1p1,  start_year: 2006,  end_year: 2026}
+preprocessors:
+  preproc:
+    regrid:
+      target_grid: MIROC5
+      scheme: linear
+    mask_fillvalues:
+      threshold_fraction: 0.95
+    extract_region:
+      start_longitude: -60
+      end_longitude: 40
+      start_latitude: 30
+      end_latitude: 70
+diagnostics:
+  weather_regime:
+    description: Compute modes of variability.
+    variables:
+      psl:
+        preprocessor: preproc
+        mip: day
+        field: T2M
+    scripts:
+      main:
+        script: magic_bsc/weather_regime.r #(in diag_scripts)
+
+        #Region considered
+        region: Polar # North-Atlantic or Polar
+
+
+        #Start and end periods for the historical and projection periods
+        start_historical: "1961-01-01"
+        end_historical: "1990-12-31"
+        start_projection: "2006-01-01"
+        end_projection: "2026-12-31"
+
+        #Regime parameters
+        ncenters: 3
+        detrend_order: 2
+        cluster_method: "kmeans" # If NULL it computes hcluster else specify kmeans
```

Comment:
If I try to use `cluster_method: null` the run ends with the following error:
```
Loading required package: maps

Attaching package: ‘s2dverification’

The following object is masked from ‘package:base’:

    Filter


Attaching package: ‘ClimProjDiags’

The following object is masked from ‘package:s2dverification’:

    Subset

Warning message:
In dir.create(run_dir, recursive = TRUE) :
  '/group_workspaces/jasmin2/cp4cds1/magic-test/tmp-bandela/MAGIC_BSC/output_dir/recipe_modes_of_variability_wp4_cluster_method_None_20190124_153545/run/weather_regime/main' already exists
 lon  lat time 
  72   29 3650 
[1] "Seasonal"
model   var  time   lon   lat 
    1     1  3650    72    29 
[1] 3650
[1] 72
[1] 29
Error in if (method == "kmeans") { : argument is of length zero
Calls: WeatherRegime ... lapply -> FUN -> splatted_f -> do.call -> <Anonymous>
Execution halted
```

Reply:
Hi @bouweandela ,
the cluser_method parameter can be 'kmeans' or 'hclust'. I've solved the conflict names.
Thanks for the feedback.
Núria